src/jsutils/instanceOf.ts

```diff
@@ -9,7 +9,7 @@ import { inspect } from './inspect.js';
 export const instanceOf: (value: unknown, constructor: Constructor) => boolean =
   /* c8 ignore next 6 */
   // FIXME: https://github.com/graphql/graphql-js/issues/2317
-  globalThis.process != null && globalThis.process.env.NODE_ENV === 'production'
+  globalThis?.process?.env?.NODE_ENV === 'production'
```

Comment:
This will be more inline with the rules of the project

```suggestion
  globalThis.process != null && globalThis.process.env != null && globalThis.process.env.NODE_ENV === 'production'
```